posthog-core/src/index.ts

```diff
@@ -63,38 +57,422 @@ export abstract class PostHogCore {
+    personProperties: Record<string, string> = {},
+    groupProperties: Record<string, Record<string, string>> = {}
+  ): Promise<JsonType | undefined> {
+    const payloads = await this.getFeatureFlagPayloadsStateless(distinctId, groups, personProperties, groupProperties)
+
+    if (!payloads) {
+      return undefined
+    }
+
+    const response = payloads[key]
+
+    // Undefined means a loading or missing data issue. Null means evaluation happened and there was no match
+    if (response === undefined) {
+      return null
+    }
+
+    return this._parsePayload(response)
+  }
+
+  async getFeatureFlagPayloadsStateless(
+    distinctId: string,
+    groups: Record<string, string> = {},
+    personProperties: Record<string, string> = {},
+    groupProperties: Record<string, Record<string, string>> = {}
+  ): Promise<PostHogDecideResponse['featureFlagPayloads'] | undefined> {
+    const payloads = (
+      await this.getFeatureFlagsAndPayloadsStateless(distinctId, groups, personProperties, groupProperties)
+    ).payloads
+
+    if (payloads) {
+      return Object.fromEntries(Object.entries(payloads).map(([k, v]) => [k, this._parsePayload(v)]))
+    }
+    return payloads
+  }
+
+  _parsePayload(response: any): any {
+    try {
+      return JSON.parse(response)
+    } catch {
+      return response
+    }
+  }
+
+  async getFeatureFlagsStateless(
+    distinctId: string,
+    groups: Record<string, string> = {},
+    personProperties: Record<string, string> = {},
+    groupProperties: Record<string, Record<string, string>> = {}
+  ): Promise<PostHogDecideResponse['featureFlags'] | undefined> {
+    return (await this.getFeatureFlagsAndPayloadsStateless(distinctId, groups, personProperties, groupProperties)).flags
+  }
+
+  async getFeatureFlagsAndPayloadsStateless(
+    distinctId: string,
+    groups: Record<string, string> = {},
+    personProperties: Record<string, string> = {},
+    groupProperties: Record<string, Record<string, string>> = {}
+  ): Promise<{
+    flags: PostHogDecideResponse['featureFlags'] | undefined
+    payloads: PostHogDecideResponse['featureFlagPayloads'] | undefined
+  }> {
+    const decideResponse = await this.getDecide(distinctId, groups, personProperties, groupProperties)
+
+    const flags = decideResponse?.featureFlags
+    const payloads = decideResponse?.featureFlagPayloads
+
+    return {
+      flags,
+      payloads,
+    }
+  }
+
+  /***
+   *** QUEUEING AND FLUSHING
+   ***/
+  protected enqueue(type: string, _message: any, options?: PosthogCaptureOptions): void {
+    const message = {
+      ..._message,
+      type: type,
+      library: this.getLibraryId(),
+      library_version: this.getLibraryVersion(),
+      timestamp: options?.timestamp ? options?.timestamp : currentISOTime(),
+    }
+
+    if (message.distinctId) {
+      message.distinct_id = message.distinctId
+      delete message.distinctId
+    }
+
+    // TODO: check why this isn't flushed properly....
```

Comment:
Still todo?

Reply:
this issue: https://github.com/PostHog/posthog/issues/13092 - I have a test I didn't commit (because it will fail :P )